front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -124,21 +133,68 @@ class PipelineController {
     if (pipeline.id != existingPipeline.id) {
       throw new InvalidRequestException("The provided id ${id} doesn't match the pipeline id ${pipeline.id}")
     }
-    pipeline.name = pipeline.getName().trim()
 
-    if (pipelineDAO.getPipelinesByApplication(pipeline.getApplication()).any {
-      it.getName().equalsIgnoreCase(pipeline.getName()) && it.getId() != id }) {
-      throw new DuplicateEntityException("A pipeline with name ${pipeline.getName()} already exists in application ${pipeline.application}")
-    }
+    validatePipeline(pipeline)
+
+    pipeline.name = pipeline.getName().trim()
 
     pipeline.updateTs = System.currentTimeMillis()
     pipelineDAO.update(id, pipeline)
     return pipeline
   }
 
-  private void checkForDuplicatePipeline(String application, String name) {
+  /**
+   * Ensure basic validity of the pipeline. Invalid pipelines will raise runtime exceptions.
+   *
+   *
+   * @param pipeline The Pipeline to validate
+   */
+  private void validatePipeline(Pipeline pipeline) {
+
+    // Pipelines must has an applicatoin and name
+    if (!pipeline.application || !pipeline.name) {
+      throw new InvalidEntityException("A pipeline requires name and application fields")
+    }
+
+    //Check if pipeline type is templated
+    if(pipeline.getType() == "templatedPipeline") {
```

Comment:
This string should probably be extracted as well.

Reply:
Add space `if (`.